### DISCUSSION TITLE
how to set up the model with unknown background  as nuisance parameter？

### DISCUSSION THREAD

**Discussion by Gongyuanlin** [scikit-hep/pyhf#2440] [Q&A]
Thanks for the pyhf team providing such a wonderful package!

I am new to pyhf and want to ask if there is  unknown background component $b_1$ in my backgrounds $b=b_1+b_2$  in my model, how can I set it as nuisance parameter? As I have learned in the pyhf tutorial, we can set up a simple model using the following code：

`model = pyhf.simplemodels.uncorrelated_background(
    signal=[5.0, 10.0], bkg=[50.0, 60.0], bkg_uncertainty=[5.0, 12.0]
)
model`

My question will be that how I can modify the code to set the $b_1$ as the nuisance parameter?

**Comment by alexander-held** [comment]
In general for arbitrary models you have to write down the specification for the model (https://pyhf.readthedocs.io/en/v0.7.6/likelihood.html) which you can then turn into a `pyhf.Model`. There is no equivalent to `pyhf.simplemodels` for arbitrary models. Another option for building models is [`cabinetry`](https://github.com/scikit-hep/cabinetry/), when using that you would also start by writing a configuration for your model that then gets constructed.

For your case the specification depends a bit on what $b_1$ exactly is. I assume you have multiple background contributions, which each would be their own sample in `pyhf`, and those two contributions have yields of $b_1$ and $b_2$. It then would depend on what kind of nuisance parameter you are looking for. If the normalization of $b_1$ is completely unknown and meant to float freely, you would attach a `normfactor` modifier to it. You can also use `normsys` or `histosys` modifiers for normalization and shape variations with a corresponding constraint term.
